programs/evm_loader/src/processor.rs

```diff
@@ -0,0 +1,231 @@
+        LoaderInstruction::Write { offset, bytes } =>
+            EVMProcessor::do_write(keyed_accounts, offset, &bytes),
+        LoaderInstruction::Finalize =>
+            EVMProcessor::do_finalize(keyed_accounts),
+        LoaderInstruction::InvokeMain { data } =>
+            EVMProcessor::do_invoke_main(keyed_accounts, &data),
+    }
+}
+
+pub struct EVMProcessor;
+
+impl EVMProcessor {
+    fn read_account(bytes: &[u8]) -> Result<(H160, MemoryAccount), InstructionError> {
+        limited_deserialize(bytes)
+    }
+
+    fn write_account(address: H160, account: MemoryAccount) -> Result<Vec<u8>, InstructionError> {
+        bincode::serialize(&(address, account)).map_err(|_| InstructionError::InvalidAccountData)
+    }
+
+    fn apply_accounts(
+        applies: impl IntoIterator<Item=Apply<impl IntoIterator<Item=(H256, H256)>>>,
+        keyed_accounts: &mut [KeyedAccount],
+    ) -> Result<(), InstructionError> {
+        for apply in applies {
+            match apply {
+                Apply::Modify {
+                    address,
+                    basic,
+                    code,
+                    storage,
+                    reset_storage,
+                } => {
+                    let mut processed = false;
+                    'inner_modify: for state_account in keyed_accounts.iter_mut() {
+                        let (keyed_address, mut keyed_account) =
+                            Self::read_account(&state_account.account.data)?;
+
+                        if keyed_address == address {
+                            keyed_account.nonce = basic.nonce;
+                            keyed_account.balance = basic.balance;
+
+                            if reset_storage {
+                                keyed_account.storage = Default::default();
+                            }
+
+                            for (key, value) in storage {
+                                keyed_account.storage.insert(key, value);
+                            }
+
+                            if let Some(code) = code {
+                                keyed_account.code = code;
+                            }
+
+                            state_account.account.data =
+                                Self::write_account(keyed_address, keyed_account)?;
+                            processed = true;
+                            break 'inner_modify
+                        }
+                    }
+
+                    if !processed {
+                        return Err(InstructionError::NotEnoughAccountKeys)
+                    }
+                },
+                Apply::Delete {
+                    address
+                } => {
+                    let mut processed = false;
+                    'inner_delete: for state_account in keyed_accounts.iter_mut() {
+                        let (keyed_address, _) =
+                            Self::read_account(&state_account.account.data)?;
+
+                        if keyed_address == address {
```

Comment:
nit: Current variable names might be misleaading.  These `keyed_address` and `keyed_account` types are not Solana `KeyedAccount` types but `H160` and `MemoryAccount`?